### PR TITLE
Standardize code style using Black

### LINKED ISSUE #755: Add precommit in workflow to standardize code style
This is not a function but more like coding/contributing related.

I know now quite many open source projects add `black` for instance in their pre-commit pipeline to kind of standardize the code style. I tried to `black` several files and found some look more 'pretty'. Not sure if it is something Koalas wants to have as well.

If you like it, I would be happy to work on this.

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR proposes to add [Black](https://github.com/psf/black) into Koalas.

- `dev/linter-python` detects if codes are formatted or not by Black
- `dev/reformat` script that reformats the codes by using black.
- It still keeps 100 line length style, and ignores `E231` and `E203` at pycodestyle. See also https://github.com/psf/black/issues/429 and https://github.com/psf/black/issues/1202. Black automatically formats and violates several rules.
- Update contribution guide.

Resolve #755

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @ueshin and @itholic

File: dev/tox.ini [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 
 [pycodestyle]
-ignore=E226,E241,E305,E402,E722,E731,E741,W503,W504
+ignore=E226,E241,E305,E402,E722,E731,E741,W503,W504,E231,E203
```
Comment (deepyaman) [inline dev/tox.ini]:
```suggestion
ignore=E203,E226,E231,E241,E305,E402,E722,E731,E741,W503,W504
```
(to be consistent with how codes are ordered throughout the file)

Comment (HyukjinKwon) [reply dev/tox.ini]: Thanks, @deepyaman

**Comment by HyukjinKwon** [comment]
I manually pushed to keep the co-author - https://github.com/databricks/koalas/commit/30b33344cdf7c41e2a2061daed314cdd55ec3215

Let's see if it's really useful ...

**Comment by itholic** [comment]
nice!

**Pull request closed without merging**